### PR TITLE
Fixed constant-time comparison in `Digest`.

### PR DESCRIPTION
**Pull request by jonasfj**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.1.3
+
```
Comment (sortie) [inline CHANGELOG.md]: Extra line?

File: lib/src/digest.dart [hunk 2]
```diff
@@ -17,8 +17,22 @@ class Digest {
   /// This should be used instead of manual comparisons to avoid leaking
   /// information via timing.
   @override
-  bool operator ==(Object other) =>
-      other is Digest && const ListEquality().equals(bytes, other.bytes);
+  bool operator ==(Object other) {
+    if (other is Digest) {
+      List<int> a = bytes;
+      List<int> b = other.bytes;
```
Comment (sortie) [inline lib/src/digest.dart]: a and b could be final?

File: lib/src/digest.dart [hunk 3]
```diff
@@ -17,8 +17,22 @@ class Digest {
...
+      if (a.length != b.length) {
+        return false;
```
Comment (sortie) [inline lib/src/digest.dart]: Document this behavior. An alternative reasonable thing is to keep running through both of them no matter how long they are although that is a little more likely to leak.

Comment (jonasfj) [reply lib/src/digest.dart]: Yeah, I considered it overkill, as I'll have to use the length of either `a` or `b`.

**Pull request merged**